lib/src/kbasesearchengine/main/IndexerWorker.java

```diff
@@ -841,18 +801,139 @@ private void indexObjectWrapperFn(final List<Object> input)
             }
             return ret;
         }
-        
-        private List<ObjectData> getObjectsByIds(final Set<GUID> guids)
-                throws RetriableIndexingException {
-            kbasesearchengine.search.PostProcessing pp = 
-                    new kbasesearchengine.search.PostProcessing();
-            pp.objectData = false;
-            pp.objectKeys = true;
-            try {
-                return indexingStorage.getObjectsByIds(guids, pp);
-            } catch (IOException e) {
-                throw new RetriableIndexingException(ErrorType.OTHER, e.getMessage(), e);
+
+        /** Note that an object may not be returned for a specified unresolvedGUID is the
+         * object parsing rules for that object are not found.
+         *
+         * @param objectRefPath
+         * @param unresolvedGUIDs
+         * @return
+         * @throws IndexingException
+         * @throws InterruptedException
+         * @throws RetriableIndexingException
+         */
+        private List<ObjectData> getObjectsByIds(final List<GUID> objectRefPath,
+                                                 final Set<GUID> unresolvedGUIDs)
+                throws IndexingException, InterruptedException, RetriableIndexingException {
+            final List<ObjectData> data = new ArrayList<>();
+
+            for(GUID unresolvedGUID: unresolvedGUIDs) {
+                // resolve reference
+                ResolvedReference resRef =
+                        resolveRefs(objectRefPath, ImmutableSet.of(unresolvedGUID)).
+                                                                     iterator().next();
+
+                // add subObjType and subObjId from the unresolved guid as it is
+                // not contained in the resolved reference
+                GUID guid = resRef.getResolvedReference();
+                guid = new GUID(guid.getStorageCode(),
+                                guid.getAccessGroupId(),
+                                guid.getAccessGroupObjectId(),
+                                guid.getVersion(),
+                                unresolvedGUID.getSubObjectType(),
+                                unresolvedGUID.getSubObjectId());
+                resRef = new ResolvedReference(resRef.getReference(),
+                                               guid,
+                                               resRef.getType(),
+                                               resRef.getTimestamp());
+
+                // get object from data source
+                final File tempFile;
+                try {
+                    FileUtil.getOrCreateSubDir(rootTempDir, guid.getStorageCode());
```

Comment:
I'm confused about what's going on here. It looks like the subdir created on this line isn't used on the next line.

Reply:
Fixed. Now it is.